staging/src/k8s.io/apiserver/pkg/storage/cacher/caching_object.go

```diff
@@ -64,6 +64,16 @@ type serializationsCache map[runtime.Identifier]*serializationResult
 type cachingObject struct {
 	lock sync.RWMutex
 
+	// deepCopied defines whether the object below has already been
+	// deep copied. The operation is performed lazily on the first
+	// setXxx operation.
+	//
+	// The lazy deep-copy make is useful, as effectively the only
+	// case when we are setting some fields are ResourceVersion for
+	// DELETE events, so in all other cases we can effectively avoid
+	// performing any deep copies.
+	deepCopied bool
```

Comment:
should set this to true in `DeepCopyObject()`, since we're populating it with a deep-copied object there

Reply:
done